### PR TITLE
style: Track the visited-handling-mode on the MatchingContext.

### PR DESCRIPTION
**Pull request by emilio**
This fixes bugs where we're not passing the value around correctly, like from
::-moz-any.

This is a fix for https://bugzilla.mozilla.org/show_bug.cgi?id=1431539.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19817)

### CODE REVIEW

File: components/selectors/context.rs [hunk 1]
```diff
@@ -181,6 +181,12 @@ where
         }
     }
 
+    /// How many times deep are we in a selector.
+    #[inline]
+    pub fn nesting_level(&self) -> usize {
```
Comment (nox) [inline components/selectors/context.rs]: Maybe consider introducing a `is_nested` method instead, given we only call this to compare it to 0.

**Review by nox — APPROVED** [review]
r=me with a single remark.

**Comment by tigercosmos** [comment]
```
    Updating git repository `https://github.com/servo/gecko-media.git`
Build FAILED in 0:00:08
error: failed to load source for a dependency on `gecko-media`
```
need to rebase to master?

**Comment by jdm** [comment]
No, just some ongoing bustage in our CI setup.

**Comment by emilio** [comment]
@bors-servo retry

 * A PR passed not long ago.

**Pull request merged**